### PR TITLE
Project

### PR DESCRIPTION
**Pull request by ULeuchyk**

### CODE REVIEW

File: api/constants/constants.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+export const userForPatchMethod = {
+    "name": "Nina",
+    "job": "Teacher"
+}
+
+export const invalidLoginCredentials = {
+    "email": "someemail@gmail.com",
+}
```
Comment (VladSelikh) [inline api/constants/constants.ts:29]: You don't need quotes around keys for these objects.

File: unit/src/registration-form.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        errorMessage += EmailError + "\n";
+      }
+      if (this.password.length < minPasswordLength || !this.password.match(passwordPattern)) {
+        errorMessage += PasswordError + "\n";
+      }
+      if (this.repeatPassword !== this.password) {
+        errorMessage += RepeatPasswordError;
+      }
+      if (errorMessage) {
+      return errorMessage = errorMessage.trim();
```
Comment (VladSelikh) [inline unit/src/registration-form.ts]: Please correct:
```
 return errorMessage.trim();
```

File: unit/src/registration-form.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+     passwordPattern } from "./constants/constants";
+
+
+export class RegistrationForm {
+
+    constructor(protected name: string, protected email: string, protected password: string, protected repeatPassword: string ) {
+  
+    }
+
+    public submitButton(): string {
```
Comment (VladSelikh) [inline unit/src/registration-form.ts]: Please name the method according to its content: `submitForm()`.

File: unit/jest-test/registrationForm.spec.ts [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  let registrationForm: RegistrationForm;
+
+  beforeEach(() => {
+    registrationForm = new RegistrationForm(
+      validName,
+      validEmail,
+      validPasswordMoreThenEightCharacters,
+      validPasswordMoreThenEightCharacters
+    );
+  });
```
Comment (VladSelikh) [inline unit/jest-test/registrationForm.spec.ts]: This precondition only fits for the first test. Please add this to the first test and delete `beforeEach()` block.

File: ui/src/pages/basePage.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        await expect(this.page).toHaveTitle(title);
+    }
+
+    public getAcceptCookiesButton() {
+        return this.page.locator(`button#onetrust-accept-btn-handler`);
+    }
+
+    public getElementByLocator(locator: string) {
+        return this.page.locator(locator);
+    }
```
Comment (VladSelikh) [inline ui/src/pages/basePage.ts]: What is the point of using this method? Why not use an ordinary getter for a separate page?

File: ui/src/support/types.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    LEADERSHIP = "Leadership ",
+    TEAM = "Team ",
+    HANDBOOK = "Handbook ",
+    INVESTOR = "Investor relations ",
+    TRUST = "Trust Center ",
+    NEWSLETTER = "Newsletter ",
+    PRESS = "Press "
+}
+
+export enum CONTACT_US_DROP_DOWN {
```
Comment (VladSelikh) [inline ui/src/support/types.ts]: As far as I can see these enum properties are not called anywhere in your code. Please delete it.

File: ui/src/pages/elements/navigationBar.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+   public getSearchButton() {
+      return this.page.locator(searchButton)
+   }
+
+
+   public async getSearchField() {
+      await (this.getSearchButton()).click();
+      return this.page.locator(`//div[@id="be-navigation-desktop"]//input[@placeholder="Search"]`);
+   }
```
Comment (VladSelikh) [inline ui/src/pages/elements/navigationBar.ts:28]: There should be no actions inside your getters. Please delete `await (this.getSearchButton()).click();` line.

File: ui/src/pages/elements/navigationBar.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+   public async searchFor(text: string) {
+      const searchField = this.getSearchField();
+      await (await searchField).type(text, { delay: defaultTimer });
```
Comment (VladSelikh) [inline ui/src/pages/elements/navigationBar.ts]: You don'e need `await` before `searchField`.

File: ui/src/pages/elements/navigationBar.ts [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+      await this.page.keyboard.press("Enter");
+   }
+
+   public getSignButton(text: BUTTON_SIGN) {
```
Comment (VladSelikh) [inline ui/src/pages/elements/navigationBar.ts]: `getSignButton` by what? Please name the method accordingly.

File: ui/src/pages/signUpPage.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    return this.page.locator(`button[type = "submit"]`)
+  }
+
+  public async fillFirstNameField(firstName: string,) {
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.FIRST_NAME)).type(firstName, { delay: defaultTimer });
+  }
+
+
+  public async performSignUp(firstName: string, lastName: string, username: string, email: string, password: string) {
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.FIRST_NAME)).type(firstName, { delay: defaultTimer });
```
Comment (VladSelikh) [inline ui/src/pages/signUpPage.ts]: You already have a method for this action. Either use it here or delete `fillFirstNameField(firstName: string,)` method.

File: ui/tests/playwright-gitlab.spec.ts [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+    })
+
+    test("Should open Sign Up page correctly", async () => {
+        await (await newsletterPage.navigationBar.getSignButton(BUTTON_SIGN.UP)).click();
+        signUpPage.waitUntilUrlContains(SIGN_UP_URL)
+    })
+})
+test.describe("GitLab SignUp tests", async () => {
+    test.beforeEach(async ({ browser }) => {
+        await (await signUpPage.getInputFieldByName(SIGN_UP_FIELDS.FIRST_NAME)).fill('')
```
Comment (VladSelikh) [inline ui/tests/playwright-gitlab.spec.ts]: The same as above.

File: api/constants/constants.ts (hunk 1 above)
Comment (ULeuchyk) [reply api/constants/constants.ts:29]: done

File: unit/src/registration-form.ts (hunk 2 above)
Comment (ULeuchyk) [reply unit/src/registration-form.ts]: done

File: unit/src/registration-form.ts (hunk 3 above)
Comment (ULeuchyk) [reply unit/src/registration-form.ts]: done

File: unit/jest-test/registrationForm.spec.ts (hunk 4 above)
Comment (ULeuchyk) [reply unit/jest-test/registrationForm.spec.ts]: done

File: ui/src/support/types.ts (hunk 6 above)
Comment (ULeuchyk) [reply ui/src/support/types.ts]: done

File: ui/src/pages/basePage.ts (hunk 5 above)
Comment (ULeuchyk) [reply ui/src/pages/basePage.ts]: done

File: ui/src/pages/elements/navigationBar.ts (hunk 7 above)
Comment (ULeuchyk) [reply ui/src/pages/elements/navigationBar.ts:28]: done

File: ui/src/pages/elements/navigationBar.ts (hunk 8 above)
Comment (ULeuchyk) [reply ui/src/pages/elements/navigationBar.ts]: done

File: ui/src/pages/elements/navigationBar.ts (hunk 9 above)
Comment (ULeuchyk) [reply ui/src/pages/elements/navigationBar.ts]: done

File: ui/src/pages/signUpPage.ts (hunk 10 above)
Comment (ULeuchyk) [reply ui/src/pages/signUpPage.ts]: done

File: ui/tests/playwright-gitlab.spec.ts (hunk 11 above)
Comment (ULeuchyk) [reply ui/tests/playwright-gitlab.spec.ts]: I think in this case it is not applicable. "fill" and "type" are different actions.

File: unit/src/registration-form.ts [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+    if (this.password.length < minPasswordLength || !this.password.match(passwordPattern)) {
+      errorMessage += PasswordError + "\n";
+    }
+    if (this.repeatPassword !== this.password) {
+      errorMessage += RepeatPasswordError;
+    }
+    if (errorMessage) {
+      return errorMessage.trim();
+    }
+    else {
```
Comment (VladSelikh) [inline unit/src/registration-form.ts]: You don't need `else` here.

File: api/constants/constants.ts [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+
+export const userForPutMethod = {
+    name: "Ivan",
+    job: "Developer"
+}
+
+export const userForPatchMethod = {
+    name: "Nina",
+    job: "Teacher"
+}
```
Comment (VladSelikh) [inline api/constants/constants.ts:25]: Please create a type for:
```
{
   name: string;
   job: string;
}
```

File: ui/src/pages/signUpPage.ts [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+  public async getInputFieldByName(field: SIGN_UP_FIELDS) {
+    return this.page.locator(`#new_user_${field}`)
+  }
+
+
+  public getContinueButton() {
+    return this.page.locator(`button[data-qa-selector = "new_user_register_button"]`)
+  }
+
+  public getLengthError() {
```
Comment (VladSelikh) [inline ui/src/pages/signUpPage.ts]: `getLengthError` what? Please provide an element type.

File: ui/src/pages/signUpPage.ts [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+    return this.page.locator(lengthErrorElement)
+  }
+
+  public getUsernameError() {
```
Comment (VladSelikh) [inline ui/src/pages/signUpPage.ts]: The same as above.

File: ui/src/support/constants.ts [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+export const newsletterPageTitle = "Contact | GitLab"
+export const searchObject = "API"
+export const validFirstName = "Vola"
+export const validLastName = "Dominos"
+export const validUsername = generateRandomString(10)
+export const validEmail = generateRandomString(5)+"@tesdfsadfsdfst.gultest"
+export const validPassword = "1234TYdas"
+export const tooLongFirstname = generateRandomString(128)
+export const alreadyTakenUsername = 'Bear'
+export const lengthErrorElement = `//div[@class ="col form-group"]//p[@class = 'gl-field-error']`
```
Comment (VladSelikh) [inline ui/src/support/constants.ts]: It is not an element, it is a locator.

File: ui/src/support/constants.ts [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+export const lengthErrorText = "First name is too long (maximum is 127 characters)" 
+export const takenUsernameErrorElement = `//li[contains(text(), 'Username has already been taken')]`
```
Comment (VladSelikh) [inline ui/src/support/constants.ts]: It is not an element, it is a locator.

File: ui/src/support/constants.ts [hunk 18]
```diff
@@ -0,0 +1,25 @@
+import { generateRandomString } from "./utils"
+
+export const BASE_URL = "https://about.gitlab.com/"
+export const SIGN_UP_URL = "https://gitlab.com/-/trial_registrations/new"
+export const defaultTimer = 50
+export const searchButton = `//button[contains(@aria-label ,'Search')]`
+export const confirmationMessage = `//div[contains(text(), "You must confirm your email within 1 days of signing up. If you do not confirm your email in this timeframe, your account will be deleted and you will need to sign up for GitLab again.")]`
```
Comment (VladSelikh) [inline ui/src/support/constants.ts]: It is a locator. Please name the constant accordingly.

File: ui/src/pages/signUpPage.ts [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+    return this.page.locator(takenUsernameErrorElement)
+  }
+
+  public getConfirmationMessage() {
+    return this.page.locator(confirmationMessage)
+  }
+
+  public async fillFirstNameField(firstName: string,) {
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.FIRST_NAME)).type(firstName, { delay: defaultTimer });
+  }
```
Comment (VladSelikh) [inline ui/src/pages/signUpPage.ts:41]: I don't see any great necessity to create a wrapper method over `await (await this.getInputFieldByName(SIGN_UP_FIELDS.FIRST_NAME)).type(firstName, { delay: defaultTimer });` action. Please delete it.

File: unit/src/registration-form.ts (hunk 12 above)
Comment (ULeuchyk) [reply unit/src/registration-form.ts]: done

File: api/constants/constants.ts (hunk 13 above)
Comment (ULeuchyk) [reply api/constants/constants.ts:25]: done

File: ui/src/support/constants.ts (hunk 16 above)
Comment (ULeuchyk) [reply ui/src/support/constants.ts]: done

File: ui/src/support/constants.ts (hunk 18 above)
Comment (ULeuchyk) [reply ui/src/support/constants.ts]: done

File: ui/src/support/constants.ts (hunk 17 above)
Comment (ULeuchyk) [reply ui/src/support/constants.ts]: done

File: ui/src/pages/signUpPage.ts (hunk 14 above)
Comment (ULeuchyk) [reply ui/src/pages/signUpPage.ts]: done

File: ui/src/pages/signUpPage.ts (hunk 15 above)
Comment (ULeuchyk) [reply ui/src/pages/signUpPage.ts]: done

File: ui/src/pages/signUpPage.ts (hunk 19 above)
Comment (ULeuchyk) [reply ui/src/pages/signUpPage.ts:41]: done

File: unit/src/registration-form.ts [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+    }
+    if (this.repeatPassword !== this.password) {
+      errorMessage += RepeatPasswordError;
+    }
+    if (errorMessage) {
+      return errorMessage.trim();
+    }
+    {
+      return `Congratulations! New user ${this.name} is created`;
+    }
```
Comment (VladSelikh) [inline unit/src/registration-form.ts]: You don't need `{}` brackets around `return` statement.

File: unit/src/constants/constants.ts [hunk 21]
```diff
@@ -0,0 +1,20 @@
...
+export const validName: string = "Uladzimir";
+export const validNameWithSpaces: string = "Brad Pitt";
+export const validEmail: string = "jscript@email.com";
+export const validPasswordMoreThenEightCharacters: string = "passWord1@";
+export const validPasswordEightCharacters: string = "passWo1@"
+export const invalidName: string = "Brad15Pitt";
+export const invalidEmail: string = "invalidemail.com";
+export const invalidShortPassword: string = "paS1@";
+export const invalidPasswordWithoutCapital: string = "password1@";
+export const invalidPasswordWithoutdigit: string = "passWord@";
```
Comment (VladSelikh) [inline unit/src/constants/constants.ts]: Please keep lower camel case notation in mind while naming variables.

File: ui/src/pages/signUpPage.ts [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+
+  public async fillSignUpFields(firstName: string, lastName: string, username: string, email: string, password: string) {
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.FIRST_NAME)).type(firstName, { delay: defaultTimer });
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.LAST_NAME)).type(lastName, { delay: defaultTimer });
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.USERNAME)).type(username, { delay: defaultTimer });
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.EMAIL)).type(email, { delay: defaultTimer });
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.PASSWORD)).type(password, { delay: defaultTimer });
+  }
+
+  public async clickContinueButtonOnSignUpForm() {
```
Comment (VladSelikh) [inline ui/src/pages/signUpPage.ts]: There is no need to add `OnSignUpForm` text to this method name, as it is method of the `SignUpPage` class.

File: ui/src/pages/elements/navigationBar.ts [hunk 23]
```diff
@@ -0,0 +1,43 @@
+import { defaultTimer, searchButtonLocator } from "../../support/constants"
+import { Page } from "@playwright/test"
+import { BUTTON_SIGN, COMPANY_DROP_DOWN, MENU_BUTTONS_WITHOUT_DROP_DOWN, MENU_BUTTONS_WITH_DROP_DOWN } from "../../support/types";
+
+
+export class NavigationBar {
+   constructor(protected readonly page: Page) { }
+
+   public async getSimpleNavigationButtonByInnerText(text: MENU_BUTTONS_WITHOUT_DROP_DOWN) {
```
Comment (VladSelikh) [inline ui/src/pages/elements/navigationBar.ts]: It is not an async method.

File: ui/src/pages/elements/navigationBar.ts [hunk 24]
```diff
@@ -0,0 +1,43 @@
...
+      return this.page.locator(`//div[@class ='navigation']//a[contains(@name ,"${text}")]`)
+   }
+
+   public async getExtendedNavigationButtonByInnerText(text: MENU_BUTTONS_WITH_DROP_DOWN) {
```
Comment (VladSelikh) [inline ui/src/pages/elements/navigationBar.ts]: It is not an async method.

File: ui/src/pages/elements/navigationBar.ts [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+      return this.page.locator(`//button[contains(@name ,"${text}")]`)
+   }
+
+   public async getDropDownButtonByInnerText(text: COMPANY_DROP_DOWN) {
```
Comment (VladSelikh) [inline ui/src/pages/elements/navigationBar.ts]: It is not an async method.

File: ui/src/pages/elements/navigationBar.ts [hunk 26]
```diff
@@ -0,0 +1,43 @@
...
+      return this.page.locator(`//span[@class='slp-text-body2'][text() = "${text}"]`)
+   }
+
+   public getSearchButton() {
+      return this.page.locator(searchButtonLocator)
+   }
+
+
+   public async getSearchField() {
```
Comment (VladSelikh) [inline ui/src/pages/elements/navigationBar.ts]: It is not an async method.

File: ui/src/pages/signUpPage.ts [hunk 27]
```diff
@@ -0,0 +1,56 @@
...
+
+export class SignUpPage extends BasePage {
+  constructor(page: Page) {
+    super(page)
+
+    this.url = SIGN_UP_URL
+  }
+
+
+  public async getInputFieldByName(field: SIGN_UP_FIELDS) {
```
Comment (VladSelikh) [inline ui/src/pages/signUpPage.ts]: It is not an async method.

File: ui/src/support/constants.ts [hunk 28]
```diff
@@ -0,0 +1,25 @@
...
+export const confirmationMessageLocator = `//div[contains(text(), "You must confirm your email within 1 days of signing up. If you do not confirm your email in this timeframe, your account will be deleted and you will need to sign up for GitLab again.")]`
+export const homePageTitle = "The DevSecOps Platform | GitLab"
+export const newsletterPageTitle = "Contact | GitLab"
+export const searchObject = "API"
+export const validFirstName = "Vola"
+export const validLastName = "Dominos"
+export const validUsername = generateRandomString(10)
+export const validEmail = generateRandomString(5)+"@tesdfsadfsdfst.gultest"
+export const validPassword = "1234TYdas"
+export const tooLongFirstname = generateRandomString(128)
```
Comment (VladSelikh) [inline ui/src/support/constants.ts]: Please keep lower camel case notation in mind while naming variables.

File: ui/tests/playwright-gitlab.spec.ts [hunk 29]
```diff
@@ -0,0 +1,71 @@
...
+        await newsletterPage.waitUntilTitleContains(newsletterPageTitle);
+    })
+
+    test("Should search item correctly", async () => {
+        await newsletterPage.navigationBar.searchFor(searchObject);
+        expect(await newsletterPage.navigationBar.getSearchResultTitle().innerText()).toContain(searchObject)
+    })
+
+    test("Should open Sign Up page correctly", async () => {
+        await (await newsletterPage.navigationBar.getSignButtonByText(BUTTON_SIGN.UP)).click();
```
Comment (VladSelikh) [inline ui/tests/playwright-gitlab.spec.ts]: You need only one `await` call here.

File: ui/tests/playwright-gitlab.spec.ts [hunk 30]
```diff
@@ -0,0 +1,71 @@
...
+        signUpPage.waitUntilUrlContains(SIGN_UP_URL)
+    })
+})
+test.describe("GitLab SignUp tests", async () => {
+    test.beforeEach(async ({ browser }) => {
```
Comment (VladSelikh) [inline ui/tests/playwright-gitlab.spec.ts]: What is `{ browser }` passed for here?

File: ui/tests/playwright-gitlab.spec.ts [hunk 31]
```diff
@@ -0,0 +1,71 @@
...
+
+    test("Should display warning message if First name has more than 127 letter", async () => {
+        await signUpPage.fillSignUpFields(tooLongFirstname, '', '', '', '')
+        expect(await signUpPage.getLengthErrorLocator().innerText()).toContain(lengthErrorText)
+    })
+
+    test("Should display error message if Username is already taken", async () => {
+        await signUpPage.fillSignUpFields(validFirstName, validLastName, alreadyTakenUsername, validEmail, validPassword)
+        await signUpPage.clickContinueButtonOnSignUpForm()
+        expect(signUpPage.getUsernameErrorLocator()).toBeVisible()
```
Comment (VladSelikh) [inline ui/tests/playwright-gitlab.spec.ts]: `toBeVisible()` is an async call, please use `await` with this statement.

File: ui/tests/playwright-gitlab.spec.ts [hunk 32]
```diff
@@ -0,0 +1,71 @@
...
+    })
+
+    test("Should register new user", async () => {
+        await signUpPage.fillSignUpFields(validFirstName, validLastName, validUsername, validEmail, validPassword)
+        await signUpPage.clickContinueButtonOnSignUpForm()
+        expect(signUpPage.getConfirmationMessage()).toBeVisible()
```
Comment (VladSelikh) [inline ui/tests/playwright-gitlab.spec.ts]: `toBeVisible()` is an async call, please use `await` with this statement.

File: unit/src/registration-form.ts (hunk 20 above)
Comment (ULeuchyk) [reply unit/src/registration-form.ts]: done

File: unit/src/constants/constants.ts (hunk 21 above)
Comment (ULeuchyk) [reply unit/src/constants/constants.ts]: done

File: ui/src/pages/signUpPage.ts (hunk 22 above)
Comment (ULeuchyk) [reply ui/src/pages/signUpPage.ts]: done

File: ui/src/pages/elements/navigationBar.ts (hunk 23 above)
Comment (ULeuchyk) [reply ui/src/pages/elements/navigationBar.ts]: done

File: ui/src/pages/elements/navigationBar.ts (hunk 24 above)
Comment (ULeuchyk) [reply ui/src/pages/elements/navigationBar.ts]: done

File: ui/src/pages/elements/navigationBar.ts (hunk 25 above)
Comment (ULeuchyk) [reply ui/src/pages/elements/navigationBar.ts]: done

File: ui/src/pages/elements/navigationBar.ts (hunk 26 above)
Comment (ULeuchyk) [reply ui/src/pages/elements/navigationBar.ts]: done

File: ui/src/pages/signUpPage.ts (hunk 27 above)
Comment (ULeuchyk) [reply ui/src/pages/signUpPage.ts]: done

File: ui/src/support/constants.ts (hunk 28 above)
Comment (ULeuchyk) [reply ui/src/support/constants.ts]: done

File: ui/tests/playwright-gitlab.spec.ts (hunk 29 above)
Comment (ULeuchyk) [reply ui/tests/playwright-gitlab.spec.ts]: done

File: ui/tests/playwright-gitlab.spec.ts (hunk 32 above)
Comment (ULeuchyk) [reply ui/tests/playwright-gitlab.spec.ts]: done

File: ui/tests/playwright-gitlab.spec.ts (hunk 31 above)
Comment (ULeuchyk) [reply ui/tests/playwright-gitlab.spec.ts]: done

File: ui/tests/playwright-gitlab.spec.ts (hunk 30 above)
Comment (ULeuchyk) [reply ui/tests/playwright-gitlab.spec.ts]: there is no need in browser. I copy paste it from our homework block beforeEach

File: ui/src/pages/elements/navigationBar.ts [hunk 33]
```diff
@@ -0,0 +1,43 @@
...
+   }
+
+
+   public getSearchField() {
+      return this.page.locator(`//div[@id="be-navigation-desktop"]//input[@placeholder="Search"]`);
+   }
+
+   public async searchFor(text: string) {
+      await (this.getSearchButton()).click();
+      await (await this.getSearchField()).type(text, { delay: defaultTimer });
```
Comment (VladSelikh) [inline ui/src/pages/elements/navigationBar.ts]: Please delete the extra `await` here.

File: ui/src/pages/signUpPage.ts [hunk 34]
```diff
@@ -0,0 +1,56 @@
...
+  public getConfirmationMessage() {
+    return this.page.locator(confirmationMessageLocator)
+  }
+
+  public async fillSignUpFields(firstName: string, lastName: string, username: string, email: string, password: string) {
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.FIRST_NAME)).type(firstName, { delay: defaultTimer });
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.LAST_NAME)).type(lastName, { delay: defaultTimer });
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.USERNAME)).type(username, { delay: defaultTimer });
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.EMAIL)).type(email, { delay: defaultTimer });
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.PASSWORD)).type(password, { delay: defaultTimer });
```
Comment (VladSelikh) [inline ui/src/pages/signUpPage.ts]: Please delete the extra `await`-s here.

File: ui/src/pages/signUpPage.ts [hunk 35]
```diff
@@ -0,0 +1,56 @@
...
+    await this.getContinueButton().click()
+  }
+
+
+  public async cleanSignUpInputFields() {
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.FIRST_NAME)).fill('')
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.LAST_NAME)).fill('')
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.USERNAME)).fill('')
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.EMAIL)).fill('')
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.PASSWORD)).fill('')
```
Comment (VladSelikh) [inline ui/src/pages/signUpPage.ts]: Please delete the extra `await`-s here.

File: ui/tests/playwright-gitlab.spec.ts [hunk 36]
```diff
@@ -0,0 +1,71 @@
...
+        newsletterPage = PageFactory.getPage(page, PAGES.NEWSLETTER) as NewsletterPage;
+        signUpPage = PageFactory.getPage(page, PAGES.SIGNUP) as SignUpPage;
+
+    })
+
+
+    test("Should display Home page title correctly", async () => {
+        await homePage.visitPage();
+        await homePage.getAcceptCookiesButton().click();
+        await (await homePage.navigationBar.getSimpleNavigationButtonByInnerText(MENU_BUTTONS_WITHOUT_DROP_DOWN.HOME)).click();
```
Comment (VladSelikh) [inline ui/tests/playwright-gitlab.spec.ts]: Please delete the extra `await` here.

File: ui/tests/playwright-gitlab.spec.ts [hunk 37]
```diff
@@ -0,0 +1,71 @@
...
+        await homePage.waitUntilTitleContains(homePageTitle);
+    })
+
+    test("Should correctly display Newsletter page from Company drop-down list", async () => {
+        await (await homePage.navigationBar.getExtendedNavigationButtonByInnerText(MENU_BUTTONS_WITH_DROP_DOWN.COMPANY)).hover();
+        await (await homePage.navigationBar.getDropDownButtonByInnerText(COMPANY_DROP_DOWN.NEWSLETTER)).click();
```
Comment (VladSelikh) [inline ui/tests/playwright-gitlab.spec.ts]: Please delete the extra `await`-s here.

File: ui/src/pages/elements/navigationBar.ts (hunk 33 above)
Comment (ULeuchyk) [reply ui/src/pages/elements/navigationBar.ts]: done

File: ui/src/pages/signUpPage.ts (hunk 34 above)
Comment (ULeuchyk) [reply ui/src/pages/signUpPage.ts]: done

File: ui/src/pages/signUpPage.ts (hunk 35 above)
Comment (ULeuchyk) [reply ui/src/pages/signUpPage.ts]: done

File: ui/tests/playwright-gitlab.spec.ts (hunk 37 above)
Comment (ULeuchyk) [reply ui/tests/playwright-gitlab.spec.ts]: done

File: ui/tests/playwright-gitlab.spec.ts (hunk 36 above)
Comment (ULeuchyk) [reply ui/tests/playwright-gitlab.spec.ts]: done

**Review by VladSelikh — APPROVED** [review]

**Pull request merged**